dlog/commitment/src/commitment.rs

```diff
@@ -380,52 +380,32 @@ where
         plnm: &DensePolynomial<Fr<G>>,
         max: Option<usize>,
     ) -> PolyComm<G> {
-        let n = self.g.len();
-        let p = plnm.coeffs.len();
-
-        // committing all the segments without shifting
-        let unshifted = if plnm.is_zero() {
-            Vec::new()
-        } else {
-            (0..p / n + if p % n != 0 { 1 } else { 0 })
-                .map(|i| {
-                    VariableBaseMSM::multi_scalar_mul(
-                        &self.g,
-                        &plnm.coeffs[i * n..p]
-                            .iter()
-                            .map(|s| s.into_repr())
-                            .collect::<Vec<_>>(),
-                    )
-                    .into_affine()
-                })
-                .collect()
-        };
+        commit_helper(&plnm.coeffs[..], &self.g[..], plnm.is_zero(), max)
+    }
 
-        // committing only last segment shifted to the right edge of SRS
-        let shifted = match max {
-            None => None,
-            Some(max) => {
-                let start = max - (max % n);
-                if plnm.is_zero() || start >= p {
-                    Some(G::zero())
-                } else if max % n == 0 {
-                    None
-                } else {
-                    Some(
-                        VariableBaseMSM::multi_scalar_mul(
-                            &self.g[n - (max % n)..],
-                            &plnm.coeffs[start..p]
-                                .iter()
-                                .map(|s| s.into_repr())
-                                .collect::<Vec<_>>(),
-                        )
-                        .into_affine(),
-                    )
-                }
-            }
-        };
+    pub fn commit_evaluations_non_hiding(
+        &self,
+        domain: D<Fr<G>>,
+        plnm: &Evaluations<Fr<G>, D<Fr<G>>>,
+        max: Option<usize>,
+    ) -> PolyComm<G> {
+        let is_zero = plnm.evals.iter().all(|x| x.is_zero());
```

Comment:
might be a good idea to implement `is_zero()` on the [ExtendedEvaluations](https://github.com/o1-labs/proof-systems/blob/mimoo/15-wire-plonk-fixes-3/utils/src/evaluations.rs#L6) trait